shell/platform/fuchsia/dart_runner/dart_runner.cc

```diff
@@ -152,13 +156,112 @@ void RunApplicationV2(
   }
 }
 
+void RunTestApplicationV2(
+    DartRunner* runner,
+    fuchsia::component::runner::ComponentStartInfo start_info,
+    std::shared_ptr<sys::ServiceDirectory> runner_incoming_services,
+    fidl::InterfaceRequest<fuchsia::component::runner::ComponentController>
+        controller,
+    fit::function<void(std::shared_ptr<DartTestComponentControllerV2>)>
+        component_created_callback,
+    fit::function<void(DartTestComponentControllerV2*)> done_callback) {
+  const int64_t start = Dart_TimelineGetMicros();
+
+  auto test_component = std::make_shared<DartTestComponentControllerV2>(
+      std::move(start_info), runner_incoming_services, std::move(controller),
+      std::move(done_callback));
+
+  component_created_callback(test_component);
+
+  // Start up the dart isolate and serve the suite protocol.
+  test_component->SetUp();
+
+  const int64_t end = Dart_TimelineGetMicros();
+  Dart_TimelineEvent("DartTestComponentControllerV2::SetUp", start, end,
+                     Dart_Timeline_Event_Duration, 0, NULL, NULL);
+}
+
 bool EntropySource(uint8_t* buffer, intptr_t count) {
   zx_cprng_draw(buffer, count);
   return true;
 }
 
 }  // namespace
 
+// "data" and "assets" are arguments that are specific to the Flutter runner.
```

Comment:
We can get rid of some of these constants after applying my comment below, but for the ones we keep, can we move them next to kDartVMArgs?